src/EnergyPlus/OutputReportTabular.cc

```diff
@@ -6990,1558 +7109,1869 @@ namespace EnergyPlus::OutputReportTabular {
+                                                                   "On-Site Thermal Sources");
+                        }
+                    }
+                    if (produceTabular) {
+                        if (state.dataResultsFramework->resultsFramework->timeSeriesAndTabularEnabled()) {
+                            state.dataResultsFramework->resultsFramework->TabularReportsCollection.addReportTable(tableBody,
+                                                                                                        rowHead,
+                                                                                                        columnHead,
+                                                                                                        "Annual Building Utility Performance Summary",
+                                                                                                        "Entire Facility",
+                                                                                                        "On-Site Thermal Sources");
+                        }
+                    }
                 }
-            }
 
-            //---- Water Loads Sub-Table
-            // As of 12/8/2003 decided to not include this sub-table to wait
-            // until water use is implemented in EnergyPlus before displaying
-            // the table. Implementing water end-uses makes sense for EnergyPlus
-            // but since they are not really implemented as of December 2003 the
-            // table would be all zeros.  Recommendation to exclude this table
-            // for now made by Glazer and Crawley.
-            // Aug 2006, adding table in with implementation of water system, BGriffith
-            rowHead.allocate(13);
-            columnHead.allocate(2);
-            columnWidth.allocate(2);
-            columnWidth = 14; // array assignment - same for all columns
-            tableBody.allocate(2, 13);
-            {
-                auto const SELECT_CASE_var(ort->unitsStyle);
-                if (SELECT_CASE_var == iUnitsStyle::JtoKWH) {
-                    columnHead(1) = "Water [m3]";
-                } else if (SELECT_CASE_var == iUnitsStyle::InchPound) {
-                    columnHead(1) = "Water [gal]";
+                //---- Water Loads Sub-Table
+                // As of 12/8/2003 decided to not include this sub-table to wait
+                // until water use is implemented in EnergyPlus before displaying
+                // the table. Implementing water end-uses makes sense for EnergyPlus
+                // but since they are not really implemented as of December 2003 the
+                // table would be all zeros.  Recommendation to exclude this table
+                // for now made by Glazer and Crawley.
+                // Aug 2006, adding table in with implementation of water system, BGriffith
+                rowHead.allocate(13);
```

Comment:
Obviously there are tons of changes in here and I'm not able to review every single line.  No red flags as I'm looking through, but most of the review will just be dependent on analyzing outputs.

Reply:
Thanks!